File: app/src/main/java/com/tammamkhalaf/myuaeguide/categories/hereDeveloper/discoverHere/LockupEntrypoint/ItemConverter.java
package com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint;

import android.os.Build;

import androidx.annotation.RequiresApi;
import androidx.room.TypeConverter;

import com.google.gson.Gson;

import java.lang.reflect.Type;

import static android.icu.lang.UCharacter.GraphemeClusterBreak.T;

public class ItemConverter {

    @RequiresApi(api = Build.VERSION_CODES.N)
    @TypeConverter
    public String fromObjectToGson(Item item){
        return new Gson().toJson(item);
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    @TypeConverter
    public Item fromGsonToObject(String str){
        return new Gson().fromJson(str,Item.class);
    }

}

File: shared/src/commonMain/kotlin/com/tammamkhalaf/kmmsharedmodule/categories/hereDeveloper/discoverHere/LockupEntrypoint/ItemConverter.kt
package com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint

import androidx.annotation.RequiresApi

class ItemConverter {
    @RequiresApi(api = android.os.Build.VERSION_CODES.N)
    @TypeConverter
    fun fromObjectToGson(item: com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Item?): String {
        return Gson().toJson(item)
    }

    @RequiresApi(api = android.os.Build.VERSION_CODES.N)
    @TypeConverter
    fun fromGsonToObject(str: String?): com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Item {
        return Gson().fromJson(str, com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.Item::class.java)
    }
}